client/scss/components/_footer.scss

```diff
@@ -1,189 +1,112 @@
-      }
+    &:not(:first-child) {
+      border-radius: 0;
     }
   }
 
-  .preview .dropdown {
-    width: 250px;
-  }
-
-  .meta {
-    float: inline-end;
-    text-align: end;
-    padding: 7px math.div($grid-gutter-width, 2);
-    font-size: 0.85em;
-
-    p {
-      margin: 0;
-      margin-inline-end: $grid-gutter-width;
-      white-space: nowrap;
-    }
-
-    a {
-      color: inherit;
+  @include media-breakpoint-up(sm) {
+    padding: theme('spacing.[2.5]');
+    margin-inline-end: 0;
 
-      &:hover {
-        color: theme('colors.text-link-default');
-      }
+    &:not(:first-child) {
+      margin-inline-start: calc(-1 * theme('borderRadius.sm'));
     }
   }
 
-  @include media-breakpoint-down(xs) {
-    margin-top: $mobile-nice-padding;
+  &.footer__container--hidden {
+    transform: translateY(100%);
 
-    .actions,
-    .preview,
-    &__container,
-    .preview .dropdown {
-      width: 100%;
-    }
-
-    .meta {
-      p {
-        white-space: normal;
-        width: 100%;
-      }
-
-      .avatar {
-        inset-inline-start: auto;
-      }
+    @include media-breakpoint-down(xs) {
+      display: none;
     }
+  }
+}
 
-    &__container {
-      &:not(:first-child) {
-        border-radius: 0;
-      }
+.footer__save-warning {
+  font-size: 0.95em;
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
 
-      &--hidden {
-        display: none;
-      }
-    }
+  .icon {
+    font-size: 1.2em;
+    margin-inline-end: 0.5em;
+  }
 
-    &__save-warning {
-      display: flex;
-      flex-direction: row;
-      justify-content: center;
-    }
+  p {
+    margin: -0.2em 0 0 0;
   }
 
   @include media-breakpoint-up(sm) {
-    margin-inline-start: calc(#{$desktop-nice-padding} - 0.75em);
-    margin-inline-end: $desktop-nice-padding;
-    width: auto;
-    position: fixed;
-    bottom: 0;
-
-    > ul {
-      display: flex;
-    }
+    margin-inline-end: 50px;
+    align-items: center;
+  }
+}
 
-    &__container {
-      padding: 0.75em;
-      margin-inline-end: 0;
+// Footer control bar for performing actions on the page
+.actions {
+  width: 100%;
 
-      &:not(:first-child) {
-        margin-inline-start: -$border-curvature;
-      }
-    }
+  @include media-breakpoint-up(sm) {
+    width: 250px;
+  }
 
-    &__save-warning {
-      margin-inline-end: 50px;
+  &--primary {
+    @include media-breakpoint-up(sm) {
+      width: 310px;
     }
   }
-}
 
-// Footer control bar for performing actions on the page
-footer .actions,
-footer .preview {
   .button {
     @apply w-leading-none w-inline-flex w-items-center;
+    font-weight: 600;
+    padding: 0 theme('spacing.3');
+    white-space: initial;
 
     .icon {
+      width: theme('spacing.4');
       margin-inline-end: theme('spacing.2');
       flex-shrink: 0;
     }
   }
 }
 
-footer .actions {
-  .button {
-    font-weight: 600;
-    white-space: initial;
+.actions .w-dropdown-button {
+  width: 100%;
+
+  > .button {
+    flex-grow: 1;
```

Comment:
I'd like to know whether there are specific considerations on deciding where such modifications to components should live, i.e. in the component that modifies it (footer in this case) vs. the component that is modified (dropdown-button). Could you clarify?